foundry.md

```diff
@@ -0,0 +1,19 @@
+Foundry Specifications
+========================
+
+### Overview
+
+Foundry's testing framework provides a series of cheatcodes so that developers can specify what situation they want to test.
+This file describes the KEVM specification of the Foundry testing framework, which includes the definition of said cheatcodes and what does it mean for a test to pass.
+
+```k
+requires "edsl.md"
+
+module FOUNDRY
+    imports EDSL
+endmodule
+```
+
+Foundry Success Predicate
+-------------
+
```

Comment:
`````suggestion
```k
requires "edsl.md"

module FOUNDRY
    imports EDSL
    imports FOUNDRY-SUCCESS
endmodule
```

Foundry Success Predicate
-------------------------

module FOUNDRY-SUCCESS
    imports EVM

    syntax Bool ::= foundry_success() [function]
 // --------------------------------------------
    rule foundry_success() => False

endmodule

`````

Reply:
We can add this default implementation now, and also wire up `gen_claims_module()` to use it.

Reply:
You'll have to disable actually verifying the foundry claim by removing `tests/foundry/foundry.k.prove` from the list of proven claims, and instead adding `tests/foundry/kompiled/timestamp` to the list. This will make sure we are generating valid definitions on CI (by still calling the kompile target), but avoids trying to call the prover, since we'll have that `foundry_test_success()` always goes to `False` at the moment.

The list to modify in the Makefile is here: https://github.com/runtimeverification/evm-semantics/blob/83aad12534197d8f0371afcb557257cad1f36ac9/Makefile#L608

We want `tests/foundry/foundry.k.check` to stay in the list (this checks that the generation process works as expected), and we want `tests/foundry/kompiled/timestamp` to be in the list (this checks that the generation process results in valid K that can go through `kompile` correctly), but we want to remove `tests/foundry/fonudry.k.prove` from the list because once we enable this all the specs we generate will be unprovable until we implement some cheat codes and the correct implementation of this predicate.